src/shogun/mathematics/graph/NGraph.cpp

```diff
@@ -1,62 +1,126 @@
-		ngraph_input_tensors.push_back(backend->create_tensor(
-		    ngraph::element::f32, tensor.get_shape()));
-	for (const auto& tensor : output_tensors)
-		ngraph_output_tensors.push_back(backend->create_tensor(
-		    ngraph::element::f32, tensor.get_shape()));
+	for (const auto& tensor: tensors)
+	{
+	    ngraph_input_tensors.push_back(backend->create_tensor(
+	        get_ngraph_type_from_enum(tensor->get_type()), 
+	        to_ngraph_shape(tensor->get_shape())));
+	}
+	
+	ngraph::ParameterVector inputs;
+	ngraph::OutputVector outputs;
+
+	for (const auto& el: m_input_output_nodes)
+	{
+		inputs.push_back(std::static_pointer_cast<ngraph::op::Parameter>(el));
+	}
+
+	for (const auto& el: output_nodes)
+	{
+		outputs.push_back(m_lookup.at(el));
+	}
+
+	auto f = std::make_shared<ngraph::Function>(outputs, inputs);
+
+	// // for (const auto& tensor : output_tensors)
+	// //     ngraph_output_tensors.push_back(backend->create_tensor(
+	// //         ngraph::element::f32, tensor.get_shape()));
+
+	auto handle = backend->compile(f);
 
-	auto handle = backend->compile(graph->get_ngraph_function());
 	handle->call_with_validate(
 	    ngraph_input_tensors, ngraph_output_tensors);
 
-	std::vector<Tensor> results;
+
+	std::vector<std::shared_ptr<Tensor>> results;
 	for (const auto& ngraph_tensor : ngraph_output_tensors)
 	{
-		results.push_back(Tensor::create_empty(
-		    ngraph_tensor->get_shape(),
-		    get_enum_from_ngraph(ngraph_tensor->get_element_type())));
-		ngraph_tensor->read(
-		    results.back().data(),
-		    results.back().get_size() * sizeof(float));
+	    results.push_back(std::make_shared<Tensor>(
+	        from_ngraph_shape(ngraph_tensor->get_shape()),
+	        get_enum_from_ngraph(ngraph_tensor->get_element_type())));
+	    ngraph_tensor->read(
+	        results.back()->data(),
+	        results.back()->size() * sizeof(float));
 	}
 
-	return results
-	*/
+	return results;
+	
+
+	// for (auto [tensor, node]: zip_iterator(tensors, m_input_output_nodes))
+	// {
+	// 	node->evaluate_tensor(tensor);
+	// }
+
+	// std::for_each(
+	//     m_operator_output_nodes.begin(), m_operator_output_nodes.end(), [](auto& op) {
+	// 		op->operator()();
+	//     });
+}
+
+std::shared_ptr<detail::RuntimeNode>
+NGraph::get_operator(const std::shared_ptr<node::Node>& node) const
+{
+	auto type = std::type_index(typeid(*node));
+	auto op_it = OperatorRegistry().find(type);
+	// std::cout << type.name() << '\n';
+	// for(const auto& el: OperatorRegistry())
+	// 	std::cout << el.first.name() << '\n';
+	if (op_it == OperatorRegistry().end())
+	{
+		error("Could not find operator for node {}", node->to_string());
+	}
+	return op_it->second();
 }
 
-void NGraph::add_input_operator(const std::shared_ptr<Node>& node)
+void NGraph::add_input_operator(const std::shared_ptr<node::Node>& node)
 {
+	
+	auto input = get_operator(node);
+	m_lookup[node] = std::static_pointer_cast<detail::ngraph::InputNGraph>(input)->build_input(node);
```

Comment:
@vigsterkr unfortunately ngraph can't infer the shape at execution time from dynamic. So need to the building of the ngraph graph at execution. So need change the top Graph abstraction..

Reply:
nvm, just have to call a different op::Parameter constructor and create dynamic tensors